pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -479,28 +518,34 @@ func (is *InstanceService) InstanceCreate(clusterName string, name string, confi
 		}
 	}
 
+	var serverTags []string
+	if clusterSpec.DisableServerTags == false {
+		serverTags = machineTags
+		// NOTE(flaper87): This is the minimum required version
+		// to use tags.
+		is.computeClient.Microversion = "2.52"
+	}
+
+	// Get image ID
+	imageID, err := getImageID(is, config.Image)
+	if err != nil {
+		return nil, fmt.Errorf("Create new server err: %v", err)
+	}
+
```

Comment:
sorry for late reply ,but this seems different to the tag purpose ? 

https://developer.openstack.org/api-ref/compute/?expanded=create-server-detail
imageRef (Optional) | body | string | The UUID of the image to use for your server instance. This is not required in case of boot from volume. In all other cases it is required and must be a valid UUID otherwise API will return 400.
-- | -- | -- | --

so this logic seems force Image to be existing, I am ok to this since our use case seems should comes from 
image but I just want to raise this up and get agreement @iamemilio @chrigl

Reply:
other than this ,lgtm

Reply:
@jichenjc You are right, that was not on my radar. Not having an image configured is a valid case if there is a bootvolume configured.
@iamemilio We could return an empty string from getImageID if the image name is also an empty string.

We need the code to get the imageID from glance over passing in the imageName, because under the covers, gophercloud does the same thing... but when using microversion >= 2.36, the route in nova which gophercloud attempts to use, is gone. Thus it lead to #268.

Reply:
ok thanks for the sharing, I don't know the root cause of #268 and this seems a reasonable change